packages/strapi-generate-api/templates/mongoose/service.template

```diff
@@ -20,19 +159,24 @@ module.exports = {
   fetchAll: (params) => {
     // Convert `params` object to filters compatible with Mongo.
     const filters = strapi.utils.models.convertParams('<%= globalID.toLowerCase() %>', params);
-    // Select field to populate.
-    const populate = <%= globalID %>.associations
-      .filter(ast => ast.autoPopulate !== false)
-      .map(ast => ast.alias)
-      .join(' ');
-
-    return <%= globalID %>
-      .find()
-      .where(filters.where)
-      .sort(filters.sort)
+    
+    // Generate stages.
+    const populate = generateLookupStage(<%= globalID %>);
+    const match = generateMatchStage(<%= globalID %>, filters);
+
+    const result = <%= globalID %>.aggregate([
+      {
+        $match: filters.where, // Direct relation filter
+      },
+      ...populate, // Nested-Population
+      ...match, // Nested relation filter
+    ])
       .skip(filters.start)
-      .limit(filters.limit)
-      .populate(populate);
+      .limit(filters.limit);
+
+    if (filters.sort) result.sort(filters.sort);
```

Comment:
I really prefer the `if (condition) { result... }` format. It's easier to maintain 👍